src/main/java/spireMapOverhaul/zones/manasurge/ManaSurgeZone.java

```diff
@@ -0,0 +1,388 @@
+    public void atBattleStartPreDraw() {
+        for (AbstractCard card : AbstractDungeon.player.drawPile.group) {
+            if (!hasManaSurgeModifier(card)) {
+                CardModifierManager.removeAllModifiers(card, false);
+            }
+        }
+        for (AbstractCard card : AbstractDungeon.player.discardPile.group) {
+            if (!hasManaSurgeModifier(card)) {
+                CardModifierManager.removeAllModifiers(card, false);
+            }
+        }
+        for (AbstractCard card : AbstractDungeon.player.hand.group) {
+            if (!hasManaSurgeModifier(card)) {
+                CardModifierManager.removeAllModifiers(card, false);
+            }
+        }
+        AbstractDungeon.player.hand.applyPowers();
+        AbstractDungeon.actionManager.addToBottom((new ApplyPowerAction(AbstractDungeon.player, AbstractDungeon.player, new ManaSurgePower(AbstractDungeon.player, 0), 0)));
+    }
+
+    @Override
+    public void modifyRewardCards(ArrayList<AbstractCard> cards) {
+        for (AbstractCard card : cards) {
+            if (card.cost != -2 && card.type != AbstractCard.CardType.CURSE && card.type != AbstractCard.CardType.STATUS) {
+                applyRandomPermanentModifier(card);
+            }
+        }
+    }
+
+    @Override
+    public void postAddButtons(ArrayList<AbstractCampfireOption> buttons) {
+        buttons.add(new EnchantOption(!CardGroup.getGroupWithoutBottledCards(AbstractDungeon.player.masterDeck.getPurgeableCards()).isEmpty()));
+    }
+
+    @Override
+    public void postCreateShopCards(ArrayList<AbstractCard> coloredCards, ArrayList<AbstractCard> colorlessCards) {
+        for (AbstractCard card : coloredCards) {
+            if (card.cost != -2 && card.type != AbstractCard.CardType.CURSE && card.type != AbstractCard.CardType.STATUS) {
+                applyRandomPermanentModifier(card);
+            }
+        }
+        for (AbstractCard card : colorlessCards) {
+            if (card.cost != -2 && card.type != AbstractCard.CardType.CURSE && card.type != AbstractCard.CardType.STATUS) {
+                applyRandomPermanentModifier(card);
+            }
+        }
+    }
+
+    @Override
+    public float modifyCardBaseCost(AbstractCard c, float baseCost) {
+        if (hasManaSurgeModifier(c)) {
+            if (isCommonModifier(c)) {
+                if (isPositiveModifier(c)) {
+                    return baseCost + 40;
+                } else {
+                    return baseCost - 20;
+                }
+            } else {
+                if (isPositiveModifier(c)) {
+                    return baseCost + 80;
+                } else {
+                    return baseCost - 40;
+                }
+            }
+        }
+        return baseCost;
+    }
+
+    @Override
+    public AbstractEvent forceEvent() {
+        return ModifiedEventRateZone.returnIfUnseen(ManaCycloneEvent.ID);
+    }
+
+    @Override
+    public float zoneSpecificEventRate() {
+        return 1;
```

Comment:
Are you sure you need this? Since you're using forceEvent, the event will definitely spawn the first time and after that, it'll just be random events anyway. This is redundant to my knowledge.

Reply:
I thought it was redundant but the example zone used both I think so I just did it in case. Removed.